dataplane/linux/vxlan_mgr.go

```diff
@@ -194,16 +200,52 @@ func (m *vxlanManager) CompleteDeferredWork() error {
 				continue
 			}
 
-			routes = append(routes, routetable.Target{
-				Type: routetable.TargetTypeVXLAN,
-				CIDR: cidr,
-				GW:   ip.FromString(vtep.Ipv4Addr),
-			})
+			if r.Type == proto.RouteType_DEFAULT {
+				defaultRoute := routetable.Target{
+					Type:  routetable.TargetTypeDefault,
+					CIDR:  cidr,
+					GW:    ip.FromString(r.Gw),
+					Proto: routetable.RTProtocolVXLANManaged,
+				}
+
+				defaultRoutes = append(defaultRoutes, defaultRoute)
+				logCtx.WithField("route", r).Debug("adding default route to list for addition")
+			} else {
+				vxlanRoute := routetable.Target{
+					Type:  routetable.TargetTypeVXLAN,
+					CIDR:  cidr,
+					GW:    ip.FromString(vtep.Ipv4Addr),
+					Proto: routetable.RTProtocolVXLANManaged,
+				}
+
+				vxlanRoutes = append(vxlanRoutes, vxlanRoute)
+				logCtx.WithField("route", vxlanRoute).Debug("adding vxlan route to list for addition")
+			}
 		}
 
-		logrus.WithField("routes", routes).Debug("VXLAN manager sending L3 updates")
 		logrus.Info("VXLAN Manager completed deferred work")
-		m.routeTable.SetRoutes(m.vxlanDevice, routes)
+		if len(vxlanRoutes) > 0 {
+			logrus.WithField("vxlanroutes", vxlanRoutes).Debug("VXLAN manager sending VXLAN L3 updates")
+			m.routeTable.SetRoutes(m.vxlanDevice, vxlanRoutes)
+		}
+
+		if len(defaultRoutes) > 0 {
+			localInterface := m.getLocalVTEP()
+			if localInterface != nil {
```

Comment:
Do we need to handle the case where the localInterface is nil?

Reply:
Yes, at least log something